p2p/transport/webrtc/udp_mux.go

```diff
@@ -0,0 +1,341 @@
+package libp2pwebrtc
+
+import (
+	"errors"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/pion/ice/v2"
+	"github.com/pion/logging"
+	"github.com/pion/stun"
+)
+
+type candidateAddr struct {
+	raddr *net.UDPAddr
+	ufrag string
+}
+
+// UDPMuxNewAddr is an implementation of the interface
+type UDPMuxNewAddr struct {
+	params ice.UDPMuxParams
+
+	closedChan chan struct{}
+	closeOnce  sync.Once
+
+	// connsIPv4 and connsIPv6 are maps of all udpMuxedConn indexed by ufrag|network|candidateType
+	connsIPv4, connsIPv6 map[string]*udpMuxedConn
+
+	addressMapMu sync.RWMutex
+	addressMap   map[string]*udpMuxedConn
+
+	// buffer pool to recycle buffers for net.UDPAddr encodes/decodes
+	pool *sync.Pool
```

Comment:
libp2p has a shared buffer pool. Better to use that one: https://github.com/libp2p/go-buffer-pool.